### PR TITLE
Persist confirm config

### LINKED ISSUE #2815: Persist confirm state
User config should be stored in dfhack-config/confirm.json and not be associated with the current fort since the choices are global

### PR DESCRIPTION
**Pull request by robob27**
Load confirm config from dfhack-config/confirm.json. Sister scripts PR: https://github.com/DFHack/scripts/pull/617

Closes #2815

### CODE REVIEW

File: plugins/lua/confirm.lua [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+        local ok, config = pcall(json.decode_file, CONFIRM_CONFIG_FILE)
+        local conf_data = ok and config or get_conf_data()
+        if ok then
+            for _, c in ipairs(conf_data) do
+                set_conf_state(c.id, c.enabled)
+            end
+        end
+        loaded = true
+    end
+end
```
Comment (myk002) [inline plugins/lua/confirm.lua:23]: splitting the save/restore logic across files seems like asking for trouble down the line. could you move this logic into gui/confirm?

Comment (myk002) [reply plugins/lua/confirm.lua:23]: note that you have to make `gui/confirm` a `--@ module=true`

Comment (robob27) [reply plugins/lua/confirm.lua:23]: I did initially try doing this but it made the confirm options pop up as soon as the game loaded and I wasn't sure the best way to stop that from happening

Comment (robob27) [reply plugins/lua/confirm.lua:23]: As soon as I add `--@ module=true`:

image

Comment (myk002) [reply plugins/lua/confirm.lua:23]: you can't just *say* you're a module. you also have to *act* like a module : )

add:
```lua
if dfhack_flags.module then
    return
end
```

before the script has any side effects (like bringing up a window)

Comment (myk002) [reply plugins/lua/confirm.lua:23]: e.g. https://github.com/DFHack/scripts/blob/b13f351b22de50210b9b71d34806ebc2b84351ca/gui/quickfort.lua#L677

Comment (robob27) [reply plugins/lua/confirm.lua:23]: Haha.. sorry. I'm still covid dumb. That's what I missed. Thanks.

**Review by myk002 — APPROVED** [review]

**Comment by robob27** [comment]
@myk002 You probably already know but just a heads up that there was the c++ changes since the approval

File: plugins/confirm.cpp [hunk 2]
```diff
@@ -612,15 +612,7 @@ DFhackCExport command_result plugin_init (color_ostream &out, vector <PluginComm
...
-    if (is_enabled != enable)
-    {
-        for (auto c : confirmations)
-        {
-            if (!c.second->apply(enable))
-                return CR_FAILURE;
-        }
-        is_enabled = enable;
-    }
+    is_enabled = enable;
```
Comment (robob27) [inline plugins/confirm.cpp:615]: Not sure if I should just check if `is_enabled != enabled` before setting it always, or only if we are doing something when the state changes. Maybe just check anyways?

Comment (myk002) [reply plugins/confirm.cpp:615]: I think this is fine.

**Review by myk002 — APPROVED** [review]

**Pull request merged**